src/JsonSchema/Constraints/UndefinedConstraint.php

```diff
@@ -122,7 +122,7 @@ protected function validateCommonProperties($value, $schema = null, $path = null
                 // Draft 4 - Required is an array of strings - e.g. "required": ["foo", ...]
                 foreach ($schema->required as $required) {
                     if (!property_exists($value, $required)) {
-                        $this->addError($required, "The property " . $required . " is required");
+                        $this->addError( (!$path) ? $required : "$path.$required" , "The property " . $required . " is required");
```

Comment:
Ok, can i have more info what code style are you using? I cant find it anywhere in the docs